src/generators/modelsGenerator.ts

```diff
@@ -701,7 +712,14 @@ function getProperties(
       property.name === "siblings"
         ? `${(objectDetails as PolymorphicObjectDetails).unionName}[]`
         : property.type;
-    return property.nullable ? `${typeName} | null` : typeName;
+
+    if (optionsBag.armLibrary || optionsBag.ignoreNullableOnOptional) {

```

Comment:
If you do something like I suggested below, you should be able to just check for `optionsBag.ignoreNullableOnOptional`. Which would be safer as you don't need to remember to check `armLibrary` every time you check for `ignoreNullableOnOptional`